modules/custom/gcnotify/src/Plugin/WebformHandler/GCNotifyEmailHandler.php

```diff
@@ -0,0 +1,203 @@
+    $webform_values = $webform_submission->getData();
+    $translation = $webform_translation_manager->getTranslationElements($webform, $langcode);
+
+    $created = $webform_submission->get('created')->value;
+
+    // hide deprecated fields from ATI template update
+    // https://github.com/open-data/ckanext-canada/blob/master/ckanext/canada/tables/ati.yaml
+    $ati_hide_fields = ['solr_core', 'organization', 'request_summary', 'number_of_pages', 'e_mail_ati_recipient', ];
+
+    $ati_data = '';
+
+    foreach($webform_values as $key => $value) {
+      $element = $webform->getElement($key);
+      if (!in_array($key, $ati_hide_fields)) {
+
+        // get translated label of form element
+        $element_label = (in_array($key, $translation, false))
+          ? $translation[$key]['#title']
+          : $webform->getElement($key)['#title'];
+
+        // generate element in pattern [key]: value
+        $ati_data .= '**' . $element_label . '**' . "\r\n";
+        if ($element['#type'] == 'webform_address' ) {
+          $ati_data .= $value['address'];
+          if (!empty($value['address_2']))
+            $ati_data .= "\r\n" . $value['address_2'];
+          $ati_data .= "\r\n" . $value['city'];
+          $ati_data .= ", " . $value['state_province'];
+          $ati_data .= ". " . $value['postal_code'];
+          $ati_data .= "\r\n" . $value['country'] . "\r\n";
+        }
+
+        elseif ($element['#type'] == 'select' )
+          $ati_data .= $element['#options'][$value];
+
+        else
+          $ati_data .= $value;
+
+        $ati_data .= "\r\n \r\n";
+      }
+    }
+
+    $personalisation = [
+      'webform_submission_sid' => $webform_submission->id(),
+      'webform_submission_created' => \Drupal::service('date.formatter')->format($created, 'medium'),
+      'webform_submission_values' => $ati_data,
+    ];
+
+    return $personalisation;
+  }
+
+  protected function sendGCNotifyEmail($recipient, $personalisation, $handler_id)
+  {
+
+    $gcnotify_settings = \Drupal\Core\Site\Settings::get('gcnotify');
+
+    if ($gcnotify_settings) {
+
+      $api_endpoint = $gcnotify_settings['api_endpoint'];
+      $authorization = $gcnotify_settings['authorization'];
+      $langcode = \Drupal::languageManager()->getCurrentLanguage()->getId();
+      $template_id = $gcnotify_settings['template_id'][$langcode];
+      $options = [
+        'json' => [
+          'email_address' => $recipient,
+          'template_id' => $template_id,
+          'reference' => \Drupal::request()->headers->get('referer'),
+          'personalisation' => $personalisation,
+        ],
+        'headers' => [
+          'Authorization' => $authorization,
+          'Content-Type' => 'application/json',
+        ],
+      ];
+
+      $client = new Client();
+
+      try {
+
+        $response = $client->post($api_endpoint, $options);
+        if ($response->getStatusCode() == '200' || $response->getStatusCode() == '201')
```

Comment:
@RabiaSajjad not sure if GC Notify ever returns 300 type returns for redirects or caches. Should we include the 300 type response codes?

Reply:
@JVickery-TBS here are the error codes returned by GC Notify https://documentation.notification.canada.ca/en/status.html#error-codes